File: wk4/java/proofs.java
// problem link: https://open.kattis.com/problems/proofs

import java.io.*;
import java.math.*;
import java.util.*;

public class proofs {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(r.readLine());
        HashSet<String> s = new HashSet<>();
        for(int i = 1; i <= n; i++) {
            String[] a = r.readLine().split(" ");
            for(int j = 0; j < a.length - 2; j++)
                if(!s.contains(a[j])) {
                    System.out.println(i);
                    return;
                }
            s.add(a[a.length - 1]);
        }
        System.out.println("correct");
    }
}

File: wk4/cpp/proofs.cpp
// problem link: https://open.kattis.com/problems/proofs

#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
	int n;
	cin >> n;
	string g;
	getline(cin, g);
	unordered_set<string> s;
	for(int i = 1; i <= n; i++) {
	    bool f = true;
	    string t;
	    cin >> t;
	    while(f && t != "->") {
	        f = s.count(t);
	        cin >> t;
	    }
	    cin >> t;
	    if(!f) {
	        cout << i;
	        return 0;
	    }
	    s.insert(t);
	}
	cout << "correct";
	    
	return 0;
}